docs/mathlib-overview.md

```diff
@@ -0,0 +1,60 @@
+# An overview of mathlib
+
+This is an overview of mathlib as it was on 12th August 2018.
```

Comment:
This is a nice writeup, but as Patrick points out it is written like an article, and should be organized in the docs so that people expect an article. I suggest putting a by-line at the top, and putting a link to it accessible from the readme that indicates that this is an article and not a quick reference.